desci-contracts/contracts/DpidAliasRegistry.sol

```diff
@@ -0,0 +1,253 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+contract DpidAliasRegistry is Initializable, OwnableUpgradeable, PausableUpgradeable {
+    // Incremented on each dPID mint
+    uint256 public nextDpid;
+
+    // dpid => codex streamID (resolve dPID)
+    mapping(uint256 => string) public registry;
+
+    // codex streamID => dpid (check for existing alias)
+    mapping(string => uint256) public reverseRegistry;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize() public initializer {
+        OwnableUpgradeable.__Ownable_init();
+        PausableUpgradeable.__Pausable_init();
+
+        // Pause to allow owner to set nextDpid before activating minting
+        _pause();
+    }
+
+    /**
+     * Resolve the codex stream ID of a given dPID
+     * @param dpid the alias to resolve
+     */
+    function resolve(uint256 dpid) public view returns(string memory) {
+        return registry[dpid];
+    }
+
+    /**
+     * Find the dPID of a given codex stream ID, if it exists
+     * @param streamId the codex stream ID to search for
+     */
+    function find(string calldata streamId) public view returns(uint256) {
+        return reverseRegistry[streamId];
+    }
+
+    /**
+     * Signal that a stream has been bound to a new dPID alias.
+     *
+     * @param dpid the new alias
+     * @param streamID the bound stream
+     */
+    event DpidMinted (
```

Comment:
consider using indexed parameters for efficient resolution / filtering https://bitsbyblocks.com/solidity-events-everything-you-need-to-know/

Reply:
Call summary: we decided to skip this as we don't rely on efficient indexing, so lowering the gas costs makes sense.